PortalSwift/Sources/PortalSwift/Core/Api/PortalApiTypes.swift

```diff
@@ -0,0 +1,500 @@
+// There's a weird bug with Swift decoding preventing this more convenient version from working.
+// Will need to research more and try to bring this in.
+// public typealias ClientResponseMetadataNamespaces = [PortalNamespace: ClientResponseNamespaceMetadataItem]
+
+public struct ClientResponse: Codable, Equatable {
+  public let id: String
+  public let custodian: ClientResponseCustodian
+  public let createdAt: String
+  public let environment: ClientResponseEnvironment?
+  public let ejectedAt: String?
+  public let isAccountAbstracted: Bool
+  public let metadata: ClientResponseMetadata
+  public let wallets: [ClientResponseWallet]
+
+  public static func == (lhs: ClientResponse, rhs: ClientResponse) -> Bool {
+    return lhs.id == rhs.id && lhs.custodian.id == rhs.custodian.id && lhs.createdAt == rhs.createdAt && lhs.environment?.id == rhs.environment?.id && lhs.ejectedAt == rhs.ejectedAt && lhs.isAccountAbstracted == rhs.isAccountAbstracted && lhs.metadata.namespaces.eip155?.address == rhs.metadata.namespaces.eip155?.address && lhs.metadata.namespaces.solana?.address == rhs.metadata.namespaces.solana?.address && lhs.wallets[0].backupSharePairs[0].backupMethod == rhs.wallets[0].backupSharePairs[0].backupMethod
+  }
```

Comment:
This is for doing a deep comparison?

Reply:
This is only needed so that Unit Tests can leverage the `Equatable` protocol to compare results to the mock result. It's not used anywhere else 🙂